ecl/hql/hqlexpr.cpp

```diff
@@ -10705,6 +10709,62 @@ StringBuffer &CHqlUnknown::toString(StringBuffer &ret)
 
 //==============================================================================================================
 
+CHqlMacro::CHqlMacro(node_operator _op, ITypeInfo * _type, IEclPackage * _package, IAtom * _name, IFileContents * _ownedContents)
+: CHqlExpressionWithType(_op, _type), package(_package), name(_name)
+{
+    name = _name;
+    contents.setown(_ownedContents);
+}
+
+CHqlMacro *CHqlMacro::makeMacro(node_operator _op, ITypeInfo * _type, IEclPackage * _package, IAtom * _name, IFileContents * _ownedContents)
+{
+    if (!_type) _type = makeVoidType();
+    return new CHqlMacro(_op, _type, _package, _name, _ownedContents);
+}
+
+bool CHqlMacro::equals(const IHqlExpression &r) const
+{
+    if (CHqlExpressionWithType::equals(r) && name==r.queryName())
+    {
+        const CHqlMacro * other = dynamic_cast<const CHqlMacro *>(&r);
+        if (other && (contents == other->contents) && (package == other->package))
+            return true;
+    }
+    return false;
+}
+
+IInterface * CHqlMacro::queryUnknownExtra(unsigned idx)
+{
+    switch (idx)
+    {
+    case 0: return contents;
+    case 1: return package;
+    }
+    return nullptr;
+}
+
+
+void CHqlMacro::sethash()
+{
+    CHqlExpression::sethash();
+    HASHFIELD(name);
+    HASHFIELD(contents);
+    HASHFIELD(package);
+}
+
+IHqlExpression *CHqlMacro::clone(HqlExprArray &newkids)
+{
+    assertex(newkids.ordinality() == 0);
+    return LINK(this);
+}
+
+StringBuffer &CHqlMacro::toString(StringBuffer &ret)
+{
+    return ret.append(name);
```

Comment:
Should this return the macro definition rather than just the name?

Reply:
I don't think this is ever called, and it is the same as the previous implementation in CHqlUnknown - so I would keep it the same.